plugin/vsnip.vim

```diff
@@ -114,32 +114,31 @@ snoremap <silent> <Plug>(vsnip-cut-text) <C-g>:<C-u>call <SID>vsnip_visual_text(
 xnoremap <silent> <Plug>(vsnip-cut-text) :<C-u>call <SID>vsnip_visual_text(visualmode())<CR>gv"_c
 
 function! s:vsnip_cut_text_normal(type) abort
-  call s:vsnip_set_text(a:type, 'd')
-  let insertmode = s:virtualedit_in_normal() ? 'i' : 'a'
-  call feedkeys(insertmode, 'n')
+  call feedkeys(s:vsnip_set_text(a:type) . '"_c', 'n')
 endfunction
 function! s:vsnip_visual_text(type) abort
-  call s:vsnip_set_text(a:type, 'y')
+  call s:vsnip_set_text(a:type)
 endfunction
-function! s:vsnip_set_text(type, copy) abort
-  let reg_v = @v
+function! s:vsnip_set_text(type) abort
+  let oldreg = [getreg('"'), getregtype('"')]
   if a:type ==# 'v'
-    let select = '`<v`>"v'
+    let select = '`<v`>'
   elseif a:type ==# 'V'
-    let select = "'<V'>\"v"
-  elseif a:type ==? ''
-    let select = '`<`>"v'
+    let select = "'<V'>"
+  elseif a:type ==? "\<C-V>"
+    let select = "`<\<C-V>`>"
   elseif a:type ==# 'char'
   \ || (a:type ==# 'line' && s:virtualedit_in_normal())
```

Comment:
This change makes this line should be removed.

Reply:
I don't understand. You mean I have to drop the second commit? It's not necessary yes, it works the same. But non-printable characters should be avoided. But I can drop the commit if you want.